### ISSUE TITLE
Validate custom-headers for partner programs

### ISSUE THREAD

**Issue by jumde** [brave/browser-android-tabs#1104]
Custom headers fetched from the server should only be accepted if the header name is `X-Brave-Partner`; otherwise log/throw an error. This prevents an attacker who controls the Brave endpoint from being able to set any headers other than `X-Brave-Partner`.

https://github.com/brave/browser-android-tabs/blob/f72b4156345b6f7a0df6840441905584677068b5/chrome/android/java/src/org/chromium/chrome/browser/init/StatsUpdater.java#L416

**Comment by srirambv** [comment]
@samartnik @jumde  does this require any QA testing? If so please update the test plan in the PR so that it can be verified on the next build

**Comment by jumde** [comment]
> @samartnik @jumde does this require any QA testing? If so please update the test plan in the PR so that it can be verified on the next build

@srirambv - Testing would be great for this issue. But, I'm not sure if there is a good way to do it short of setting up a proxy rule to serve `X-Brave-Header` for the current partners. @SergeyZhukovsky - Do you have any ideas?

**Comment by kjozwiak** [comment]
@aekeus added a `brave.com` custom header on staging [1] that we used for https://github.com/brave/brave-browser/issues/3301#issue-408638670. However, we'll need a new `apk` that's pointing to staging to go through the same case. Unfortunately it takes a bit of time to get new builds and we want to release this today. Talked to @SergeyZhukovsky and we decided that going to the different partners and making sure we're still sending the correct headers should be good enough. Went through the following:

* visited coinbase.com and ensured that `x-brave-partner: coinbase` is being sent
* visited marketwatch.com and ensured that `X-Brave-Partner: dowjones` is being sent
* visited barrons.com and ensured that `x-brave-partner: dowjones` is being sent
* visited cheddar.com and ensured that `x-brave-partner: cheddar` is being sent
* visited brave.com and ensured there isn't any `x-brave-partner` headers being sent
* visited twitter.com and ensured there isn't any `x-brave-partner` headers being sent

When we get a build with the newest version that's pointing to staging, we'll go through the other case similar to https://github.com/brave/brave-browser/issues/3301#issue-408638670.

[1] https://laptop-updates-pre.brave.com/promo/custom-headers
